SQL/proc_descriptions/pgtap.md

```diff
@@ -0,0 +1,207 @@
+# Notes regarding pgTap testing framework
+
+## Instalation
+1) pgTAP must be installed on a host with PostgreSQL server running; it cannot be installed remotely. If you’re using PostgreSQL in Docker, you need to install pgTAP inside the Docker container.
+    If you are using Linux, you may (depending on your distribution) be able to use you distribution’s package management system to install pgTAP. For instance, on Debian, Ubuntu, or Linux Mint pgTAP can be installed with the command:
+    `sudo apt-get install pgtap`
+    On other systems pgTAP has to be downloaded and built. First, download pgTAP from PGXN (click the green download button in the upper-right). Extract the downloaded zip file, and (at the command line) navigate to the extracted folder.
+
+    To build pgTAP and install it into a PostgreSQL database, run the following commands:
+
+    ``` sh
+    make
+    make install
+    make installcheck
+    ```
+
+## Useful Notes
+
+## Notes according my work
+1) All tests could be written in one file with extension .sql (maybe tests for every procedure would be seperated in different ) and then one shell script could be executed for testing all of them (maybe even somehow convert it into a function or procedure to test in psql console). P.S. __actually__ it could be done by _Additional functions( useful simplification )_
+2) Rather than casually checking return value of the procedure to be the one expected, most of the defined procedures do not return values, thus leaving the only option as to look at the procedure: 
+    1. figure out what could be changed, 
+    2. save before-call values of the modified tables/rows/columns/cells, 
+    3. call the procedure in a sub envrionment, 
+    4. check results to be expected,
+    5. rollback.
```

Comment:
Really good idea, I'll rewrite tests according to this comment

Reply:
The one problem is that functions/procedures, which are tested, are running within __public__ schema. The only thing i've come up is parsing these functions and adding prefix `test_scheme.` to every function/procedure/table/column and evaluating resulting string in testing block

Reply:
@F-I-D-O, here is the proposition of ChatGPT 3.5:

To achieve this, you can create a schema search path in PostgreSQL. The schema search path determines the order in which schemas are searched when an unqualified object name (like a table name) is referenced in a SQL statement.

You can temporarily set the schema search path to `test_schema` before calling the function you want to test, ensuring that all references to tables, columns, and rows within the function resolve to the `test_schema`.

Here's how you can do it:

```sql
-- Set the search path to test_schema
SET search_path TO test_schema;

-- Call the function you want to test
SELECT your_function_name(parameters);

-- Reset the search path to its original value (public)
RESET search_path;
```

Replace `your_function_name(parameters)` with the actual name of the function you want to test and its parameters.

By setting the search path to `test_schema` before calling the function, PostgreSQL will search for tables, columns, and rows in the `test_schema` first. This allows you to test the function in the `test_schema` without modifying the original function.

After testing, you can reset the search path to its original value to avoid affecting subsequent statements.

Reply:
What do you think?

Reply:
If sounds fine, if it works, be free to use it

Reply:
ok